### PR TITLE
DOC: reverse rolling window (#38627)

### LINKED ISSUE #38627: DOC: example for doing a reversed rolling window
https://github.com/pandas-dev/pandas/issues/34226#issuecomment-690887774 is the example, should just add to window.rst

also would be good to show using https://pandas.pydata.org/pandas-docs/dev/reference/api/pandas.api.indexers.FixedForwardWindowIndexer.html?highlight=forward#pandas.api.indexers.FixedForwardWindowIndexer

### PR DESCRIPTION
**Pull request by mdhsieh**
This adds the suggested changes from closed pull request (#38627):
https://github.com/pandas-dev/pandas/pull/39091/files/80819f65402e46caa29ff7192b34d43f91c1a0ee
After updated commit.
- remove noqa
- change line back to single quotes
- rename `df1` to `reversed_df`
- add reference to `FixedForwardWindowIndexer`
- remove repeated `df` initialization in code example
- rename `df2` to `reversed_df`
- move code example

- [x] closes #38627
- [ ] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: doc/source/user_guide/window.rst [hunk 1]
```diff
@@ -50,16 +50,16 @@ As noted above, some operations support specifying a window based on a time offs
 
 .. ipython:: python
 
-   s = pd.Series(range(5), index=pd.date_range('2020-01-01', periods=5, freq='1D'))
-   s.rolling(window='2D').sum()
+   s = pd.Series(range(5), index=pd.date_range("2020-01-01", periods=5, freq="1D"))
+   s.rolling(window="2D").sum()
```
Comment (mzeitlin11) [inline doc/source/user_guide/window.rst:54]: Looks like there are still some nonessential changes

File: doc/source/user_guide/window.rst [hunk 2]
```diff
@@ -50,16 +50,16 @@ As noted above, some operations support specifying a window based on a time offs
...
 
 Additionally, some methods support chaining a ``groupby`` operation with a windowing operation
 which will first group the data by the specified keys and then perform a windowing operation per group.
 
 .. ipython:: python
 
-   df = pd.DataFrame({'A': ['a', 'b', 'a', 'b', 'a'], 'B': range(5)})
-   df.groupby('A').expanding().sum()
+   df = pd.DataFrame({"A": ["a", "b", "a", "b", "a"], "B": range(5)})
+   df.groupby("A").expanding().sum()
```
Comment (mzeitlin11) [inline doc/source/user_guide/window.rst:62]: same here and other places

File: doc/source/user_guide/window.rst [hunk 3]
```diff
@@ -381,8 +394,8 @@ two :class:`Series` or any combination of :class:`DataFrame`/:class:`Series` or
   with the passed Series, thus returning a DataFrame.
 * :class:`DataFrame`/:class:`DataFrame`: by default compute the statistic for matching column
   names, returning a DataFrame. If the keyword argument ``pairwise=True`` is
-  passed then computes the statistic for each pair of columns, returning a :class:`DataFrame` with a
-  :class:`MultiIndex` whose values are the dates in question (see :ref:`the next section
+  passed then computes the statistic for each pair of columns, returning a
+  ``MultiIndexed DataFrame`` whose ``index`` are the dates in question (see :ref:`the next section
```
Comment (mzeitlin11) [inline doc/source/user_guide/window.rst:398]: This change looks suspect, think you might need to merge master if you didn't add this change

File: doc/source/user_guide/window.rst [hunk 4]
```diff
@@ -592,7 +601,7 @@ The following formula is used to compute exponentially weighted mean with an inp
 
 .. math::
 
-    y_t = \frac{\sum_{i=0}^t 0.5^\frac{t_{t} - t_{i}}{\lambda} x_{t-i}}{\sum_{i=0}^t 0.5^\frac{t_{t} - t_{i}}{\lambda}},
```
Comment (mzeitlin11) [inline doc/source/user_guide/window.rst:595]: same comment w respect to merging master as above

**Review by mzeitlin11 — COMMENTED** [review]
Thanks for the pr @mdhsieh! Some initial comments

File: doc/source/user_guide/window.rst (hunk 2 above)
Comment (mdhsieh) [reply doc/source/user_guide/window.rst]: Ok, changed back

File: doc/source/user_guide/window.rst (hunk 4 above)
Comment (mdhsieh) [reply doc/source/user_guide/window.rst:595]: Ok merged

Comment (mzeitlin11) [reply doc/source/user_guide/window.rst:595]: if you check the diff, looks unchanged. You might need to fetch first, see https://pandas.pydata.org/docs/development/contributing.html#working-with-the-code for a workflow on updating a pr

Comment (mdhsieh) [reply doc/source/user_guide/window.rst]: Okay, I added the changes from master branch file since fetching and merging didn't change those lines..

**Review by mroeschke — CHANGES_REQUESTED** [review]
Could you just add the new section and save cleanups for another PR (so just add content)?

It still appears that things changed that were not supposed to in the PR.

**Comment by mdhsieh** [comment]
> Could you just add the new section and save cleanups for another PR (so just add content)?
> 
> It still appears that things changed that were not supposed to in the PR.

Yeah, I can do that.

**Comment by mdhsieh** [comment]
Yeah I realized I accidentally deleted part of a code example.

```
For some problems...
.. ipython:: python

   from pandas.api.indexers import FixedForwardWindowIndexer
   indexer = FixedForwardWindowIndexer(window_size=2)
   df.rolling(indexer, min_periods=1).sum()
```

**Comment by mdhsieh** [comment]
Updated, does this work now?

File: doc/source/user_guide/window.rst [hunk 5]
```diff
@@ -299,6 +299,38 @@ forward-looking rolling window, and we can use it as follows:
...
+           [pd.Timestamp("2018-01-01 00:00:04"), 111],
+       ],
+       columns=["time", "value"],
+   ).set_index("time")
+   df
+
+   reversed_df = df[::-1].rolling("2s").sum()[::-1]
+   reversed_df
+
+Or we can also do it using :meth:`api.indexers.FixedForwardWindowIndexer` which basically creates window boundaries for fixed-length windows that include the current row.
```
Comment (mroeschke) [inline doc/source/user_guide/window.rst:325]: This example is exactly the same as the one above this new section?

File: doc/source/user_guide/window.rst [hunk 6]
```diff
@@ -299,6 +299,38 @@ forward-looking rolling window, and we can use it as follows:
    indexer = FixedForwardWindowIndexer(window_size=2)
    df.rolling(indexer, min_periods=1).sum()
 
+Reverse rolling window
```
Comment (mroeschke) [inline doc/source/user_guide/window.rst:302]: If this example has the same result as the `FixedForwardWindowIndexer` you added in this section, you can just continue in the section above and show that it's equivalent to the code block you added here.

File: doc/source/user_guide/window.rst (hunk 5 above)
Comment (mdhsieh) [reply doc/source/user_guide/window.rst]: Right the `FixedForwardWindowIndexer` example is the same, my bad. Deleted it.

File: doc/source/user_guide/window.rst (hunk 6 above)
Comment (mdhsieh) [reply doc/source/user_guide/window.rst]: Okay, sure.

File: doc/source/user_guide/window.rst [hunk 7]
```diff
@@ -299,6 +299,24 @@ forward-looking rolling window, and we can use it as follows:
    indexer = FixedForwardWindowIndexer(window_size=2)
    df.rolling(indexer, min_periods=1).sum()
 
+We can also achieve this by using slicing in python, applying rolling aggregation, and then flipping the result as shown in example below:
```
Comment (mroeschke) [inline doc/source/user_guide/window.rst:302]: Nit: Don't need `in python`

**Review by mroeschke — COMMENTED** [review]
One minor comment otherwise LGTM

File: doc/source/user_guide/window.rst (hunk 7 above)
Comment (mdhsieh) [reply doc/source/user_guide/window.rst]: Sure, done.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @mdhsieh. CI failure unrelated

**Pull request merged**